specifyweb/frontend/js_src/lib/components/UserPreferences/Renderers.tsx

```diff
@@ -237,17 +231,17 @@ export const WelcomePageModePreferenceItem: PreferenceItemComponent<WelcomePageM
           value={value}
           onChange={handleChange}
         />
-        {value === 'customImage' || value === 'embeddedWebpage' ? (
-          <DefaultPreferenceItemRender
-            category="welcomePage"
-            definition={sourceDefinition}
+        {value === 'customImage' && (
+          <AttachmentPicker
+            displayScale={false}
             isReadOnly={isReadOnly}
-            item="source"
-            subcategory="general"
-            value={source}
-            onChange={setSource}
+            url={url === 'default' ? undefined : url}
+            onChange={(url): void => {
+              setUrl(url);
+              if (url === 'default') handleChange('default');
```

Comment:
after you modify `handleChange('default')` to `handleChange(undefined)` in `<AttachmentPicker>` don't forget to change this `if` condition here